src/rgw/rgw_formats.cc

```diff
@@ -287,7 +287,10 @@ void RGWFormatter_Plain::dump_value_int(const char *name, const char *fmt, ...)
 class HTMLHelper : public XMLFormatter {
 public:
   static std::string escape(const std::string& unescaped_str) {
-    return escape_xml_str(unescaped_str.c_str());
+    int len = escape_json_attr_len(unescaped_str.c_str(), unescaped_str.size());
+    std::string escaped(len, 0);
+    escape_json_attr(unescaped_str.c_str(), unescaped_str.size(), escaped.data());
+    return escaped;
```

Comment:
this should be calling the xml functions, not json

Reply:
Must be a unforgivable mistake, thanks!